### DISCUSSION TITLE
Configuring Elsa to be performant

### DISCUSSION THREAD

**Discussion by tsoG2000** [elsa-workflows/elsa-core#1623] [Q&A] [answered]
What kind of configuration would you recommend to us. These are our requirements:
- 30k started workflows daily that are deleted (only successfully finished ones) at the end of the day.
- Must not have duplicate workflows for certain id (our unique id representing the document)
- Must not loose dispatched workflows if server goes down etc.
- Should be highly performant
- Must not fill server disk with unnecessary data (like .ldf file that grows rapidly)
- Should not stop working if app pool tries to go into a "sleep mode"

We use are hosting Elsa server on IIS and MS SQL for database.

**Comment by sfmskywalker** [comment, marked as answer]
* Use the new [Retention](https://github.com/elsa-workflows/elsa-core/discussions/1619#discussioncomment-1465057) feature to automatically delete finished workflows (don't use it if you want to keep faulted workflows - we would need to expose this as a setting to allow you to configure this).
* Use workflow correlation. This is built-in.
* Use a persistent service bus queue such as RabbitMQ, Azure Service Bus or even SQL Server when using Rebus, or Hangfire using SQL Server.
* Design your workflows with fast-executing activities and move heavy-lifting to background workers (e.g. using Hangfire).
* See first bullet. If LDF keeps growing, [see this](https://dba.stackexchange.com/questions/29829/why-does-the-transaction-log-keep-growing-or-run-out-of-space).
* Use IIS 8 or a container service such as Kubernetes, Azure App Services (running a container) or Azure Container Services.

**Comment by chandsalam1108** [comment]
HI @sfmskywalker ,

We're using Windows Server 2012 R2, IIS 8.5 & .Net 6 with elsa 2.8.0. Please confirm if that compatible or I need to install something else.

Thank you!
